### PR TITLE
HAL: Relative URIs in embedded docs are now resolved with the self link.

### PR DESCRIPTION
**Pull request by evert**
This replaces #393.

It was pointed out by @reda-alaoui that if a HAL Form is embedded inside
an embedded document, the 'target' attribute (which is a URI) is
resolved with the URI of the main document, not the self link of an
embedded document.

I tried to find a source for what the correct behavior for _any_ link in
an embedded document HAL-Forms or HAL itself is, but there's no guidance.

Intuitively it feels that resources that are serialized are likely
serialized by the same process, whether they are stand-alone or
embedded, so if I had to pick a behavior, it makes more sense to me that
we use the embedded uri.

To my surprise and sadness, it looks like we've never done this for
HAL-Forms inside `_embedded`, but also not `_links` or nested
`_embedded` in `_embedded`.

This PR fixes this for all cases, but the question now is: "Is this a major
BC break?".

I partially feel no, because I feel that most people very likely don't
embed items from other absolute domains, and relative uris in resources
are very likely to at least always start with a `/`. I suspect that this
is the 99% case, and for those people this change is irrelevant.

But for the other 1%, this will break their clients and it means this small
change needs to be released as Ketting 8.

### CODE REVIEW

File: test/unit/state/hal-forms.ts [hunk 1]
```diff
@@ -54,6 +54,41 @@ describe('HAL forms', () => {
 
     expect(deleteAction).to.eql(expectedDeleteAction);
 
+  });
+  it('should relate relative URIs of embedded HAL forms to the embedded resource\'s self uri', async () => {
```
Comment (reda-alaoui) [inline test/unit/state/hal-forms.ts:58]: Tests added to #393 verify the behaviour when no `target` is provided. 
Could a test covering the missing `target` use case be added to this PR? :)

Comment (evert) [reply test/unit/state/hal-forms.ts:58]: Done

**Review by reda-alaoui — APPROVED** [review]
Thank you !

**Comment by reda-alaoui** [comment]
Hey, can we do something about this? :)
Tell me if you need any help.

**Comment by evert** [comment]
I'll release this as Ketting 7.1 and hopefully we don't get bug reports =)

**Comment by reda-alaoui** [comment]
Thanks :)

**Pull request merged**